src/QueryM.purs

```diff
@@ -1141,10 +1142,20 @@ mkRequestAff' listeners' webSocket logger jsonWspCall getLs inp = do
               _ -> cont (lmap dispatchErrorToError result)
         )
       respLs.addRequest id (sBody /\ getRequestInputToStore inp)
-      _wsSend webSocket (logger Debug) sBody
+      wsStatus <- _wsIsOpen webSocket
```

Comment:
I'm not familiar with how WebSockets can fail in purescript, but, this pattern seems similar to : 

- Check if file exist 
- Open file 

And the usual advice is not to do that since files can be deleted or moved after the check and before we try to open.

Reply:
In general, I do agree with the sentiment. However, there really isn't a way to avoid it since this is the same logic that is being used in the JavaScript libraries that we depend on: [ws](https://github.com/websockets/ws/blob/00c34d726dca1c558fe5ee5e346979159b2297fe/lib/websocket.js#L438-L453) and [reconnecting-websocket](https://github.com/pladaria/reconnecting-websocket/blob/ef6e6a7fc9d5078753cf36ba270e382af1f11d9e/reconnecting-websocket.ts#L252-L255)

Reply:
really? all my experience with sockets is in python and there we would try and catch. It's fine then.